CONSOLE-3777: Update NodeJS v14 to v18

cyril-ui-developer opened the pull request:
N.B. Seems this update/change would require building an image and pushing it to quay.io based on the instructions in [Docker.builder](https://github.com/openshift/console/blob/master/Dockerfile.builder#L5) file.

jhadvig reviewed (commented):
Looks good 👍 
@cyril-ui-developer please update the README accordingly

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
@jhadvig I have another [PR](https://github.com/openshift/release/pull/44053) in the release repo to update the mapping for `quay.io/coreos/tectonic-console-builder` image from v25 to v26. PTAL

cyril-ui-developer commented:
/

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

jhadvig commented:
/retest

jhadvig reviewed (approved):
/lgtm
/approve

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

cyril-ui-developer commented on README.md:
```diff
@@ -108,6 +108,25 @@ In order to enable the monitoring UI and see the "Observe" navigation item while
+   Note: You can test the image using `./builder-run.sh ./build-backend.sh`.
+         Also, to update the image on quay.io, you need edit permission to the quay.io/coreos/tectonic-console-builder repo. 
+4. Lastly, update the mapping of `tectonic-console-builder` image tag in [release-repo](https:// github.com/openshift/release/blob/master/core-services/image-mirroring/supplemental-ci-images/mapping_supplemental_ci_images_ci).
+   Note: There could be scenario were you would have to add the new image reference in the "mapping_supplemental_ci_images_ci" file, i.e. to avoid CI downtime for upcoming release cycle. 
```
@jhadvig Thanks for feedback. But I have two questions. 
1. Do we have to remove the old version mapping  at some point in time in the future. Asking bcos there is no `v24` in the [release repo mapping file](https://github.com/cyril-ui-developer/release/blob/2a835fcb0881cae3ca81f4fff6e247d63d38a54f/core-services/image-mirroring/supplemental-ci-images/mapping_supplemental_ci_images_ci#L111).
2. The [supplimental_ci_mapping file](https://github.com/openshift/release/blob/master/clusters/app.ci/supplemental-ci-images/origin-stable/release.yaml) has  reference for `"quay.io/coreos/tectonic-console-builder:v24"` but don't have for `v25`. Do we need to update this file as part of the changes in this PR?

Could you please clarify?

petr-muller replied:
sry I'm out of CI matters for too long now to correctly advise on this :(

cyril-ui-developer commented:
N.B. I am seeing the error below in the browser console:
`WebSocketClient.js:13 WebSocket connection to 'ws://localhost:8080/ws' failed: 
WebSocketClient @ WebSocketClient.js:13
initSocket @ socket.js:27
(anonymous) @ socket.js:51
index.js:493 [webpack-dev-server] Event {isTrusted: true, type: 'error', target: WebSocket, currentTarget: WebSocket, eventPhase: 2, …}` 

Looking into it....

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

jhadvig commented:
Hmm not sure why the tectonic builder contains old version but after pulling `quay.io/coreos/tectonic-console-builder:v26` locally I see v18.18.1
/retest

jhadvig commented:
OK I see the issue. It looks like `registry.ci.openshift.org/ocp/builder:rhel-8-base-nodejs-openshift-4.15` contains node v16. We need bump that version.

cyril-ui-developer commented:
> registry.ci.openshift.org/ocp/builder:rhel-8-base-nodejs-openshift-4.15

@jhadvig I have been observing the failure but not sure what is going on. If the issue is `registry.ci.openshift.org/ocp/builder:rhel-8-base-nodejs-openshift-4.15` contains node v16. was it update to node.jsv16 recently, and how come the current builder image -`tectonic-console-builder:v25` based on node.js-v14 runs without failure? 

I am wondering why [release.yaml](https://github.com/openshift/release/blob/master/clusters/app.ci/supplemental-ci-images/origin-stable/release.yaml#L100) is missing `quay.io/coreos/tectonic-console-builder:v25"` which could have been the most recent update based on builder image in quay.io prior to my PR change.. However seems it doesn't impact console UI tests since we are not experiencing any docker build failure.

I am thinking we need to update  [release.yaml](https://github.com/openshift/release/blob/master/clusters/app.ci/supplemental-ci-images/origin-stable/release.yaml#L100) with the latest builder image tag - v26 but not sure if that would fix the issue.

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

jhadvig commented:
/retest

jhadvig commented:
/retest

cyril-ui-developer commented:
/retest

jhadvig commented on README.md:
```diff
@@ -108,6 +108,25 @@ In order to enable the monitoring UI and see the "Observe" navigation item while
+   - .ci-operator.yaml
+   - Dockerfile.dev
+   - Dockerfile.plugins.demo
+   - Dockerfile.plugins.demo2
```
nit: this dockerfile is not using `tectonic-console-builder` image

jhadvig reviewed (approved):
/lgtm
/approve

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/test e2e-gcp-console

jhadvig commented:
/test e2e-gcp-console

cyril-ui-developer commented:
/test e2e-gcp-console

rhamilto commented:
@cyril-ui-developer, [Protractor and webdriver have been removed](https://github.com/openshift/console/pull/13284). :)

jhadvig commented:
/retest

cyril-ui-developer commented:
> @cyril-ui-developer, [Protractor and webdriver have been removed](https://github.com/openshift/console/pull/13284). :)

Awesome! Thank you for the heads up.

cyril-ui-developer commented:
/test e2e-gcp-console

cyril-ui-developer commented:
/retest

jcaianirh commented:
/retest

rhamilto commented:
@cyril-ui-developer, looks like there is an issue manifesting as an issue with the Cypress config resulting from these changes.  https://search.ci.openshift.org/?search=Your+configFile+threw+an+error+from%3A+cypress.config.js&maxAge=168h&context=1&type=all&name=pull-ci-openshift-console-master-e2e-gcp-console&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

cyril-ui-developer commented:
> @cyril-ui-developer, looks like there is an issue manifesting as an issue with the Cypress config resulting from these changes. https://search.ci.openshift.org/?search=Your+configFile+threw+an+error+from%3A+cypress.config.js&maxAge=168h&context=1&type=all&name=pull-ci-openshift-console-master-e2e-gcp-console&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

Correct, I have seen it. It is related to the `hashFunction` issue which I fixed in webpack.config. Added the same fix to cypress-config. Retesting to see if that fix it. It looks good in local. Thanks.

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

jcaianirh commented:
Work is tech debt and has no customer-facing impact.

/label docs-approved
/label px-approved
/label qe-approved

rhamilto commented:
/lgtm

cyril-ui-developer commented:
/test e2e-gcp-console

cyril-ui-developer commented:
/retest

cyril-ui-developer commented:
/retest

rhamilto commented:
/retest

rhamilto commented:
/retest

jcaianirh commented:
/hold cancel
